src/vcpkg/postbuildlint.cpp

```diff
@@ -973,6 +974,99 @@ namespace vcpkg::PostBuildLint
         return LintStatus::SUCCESS;
     }
 
+    static LintStatus check_no_absolute_paths_in(const Filesystem& fs, const Path& dir, Span<Path> absolute_paths)
+    {
+        static constexpr StringLiteral extensions[] = {"h", "hpp", "hxx", "py", "sh", "cmake", "pc", "cfg", "conf"};
+        std::vector<std::pair<Path, std::string>> files_and_contents;
+        for (auto& path : fs.get_regular_files_recursive(dir, IgnoreErrors{}))
+        {
+            if (Util::contains(extensions, path.extension().substr(1 /* ignore dot */)))
+            {
+                auto contents = fs.read_contents(path, VCPKG_LINE_INFO);
+                files_and_contents.emplace_back(std::move(path), std::move(contents));
+            }
+            else if (path.extension().empty())
+            {
+                auto contents = fs.read_contents(path, VCPKG_LINE_INFO);
+                if (Strings::starts_with(contents, "#!"))
+                {
+                    files_and_contents.emplace_back(std::move(path), std::move(contents));
+                }
+            }
+        }
+        std::vector<std::string> string_paths;
+        for (const auto& path : absolute_paths)
+        {
+            string_paths.push_back(path.native());
+#if defined(_WIN32)
+            string_paths.push_back(path.generic_u8string());
+#endif
```

Comment:
```suggestion
#if defined(_WIN32)
            auto path_preferred = path;
            path_preferred.make_preferred();
            string_paths.push_back(path_preferred.generic_u8string());
            string_paths.push_back(std::move(path).native());
#else
            string_paths.push_back(path.native());
#endif
```

this allows `absolute_paths` to contain paths which are not in preferred format on Windows.